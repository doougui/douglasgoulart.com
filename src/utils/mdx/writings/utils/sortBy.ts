import { Writing } from 'types/Writing';
import { parseToDate } from 'utils/time/parseToDate';
import { SortTypesKeys } from '../types';

function byNew(data: Writing[]) {
  return data.sort((a, b) => {
    const aAsDate = parseToDate(a.publishedAt);
    const bAsDate = parseToDate(b.publishedAt);
    return Number(bAsDate) - Number(aAsDate);
  });
}

function byName(data: Writing[]) {
  return data.sort((a, b) => (a.title > b.title ? 1 : -1));
}

export function sortBy(data: Writing[], type: SortTypesKeys): Writing[] {
  const sortTypes: { [key in SortTypesKeys]: () => Writing[] } = {
    new: () => byNew(data),
    name: () => byName(data),
  };

  return sortTypes[type]();
}
