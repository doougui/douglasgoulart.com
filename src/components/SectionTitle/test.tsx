import { render, screen } from 'utils/tests';

import { SectionTitle } from '.';

describe('<SectionTitle />', () => {
  it('should render correctly', () => {
    const { container } = render(<SectionTitle>SectionTitle</SectionTitle>);

    expect(
      screen.getByRole('heading', { name: /SectionTitle/i }),
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
