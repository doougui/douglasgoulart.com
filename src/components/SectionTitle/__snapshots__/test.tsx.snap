// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionTitle /> should render correctly 1`] = `
.c0 {
  font-size: 2.4rem;
  line-height: 3rem;
  color: #EDEDEF;
  font-weight: 600;
}

<h2
  class="c0"
>
  SectionTitle
</h2>
`;
